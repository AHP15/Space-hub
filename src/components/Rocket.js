import React from 'react';
import PropTypes from 'prop-types';

const Rocket = ({ name, image, description }) => (
  <div className="card">
    <img src={image} alt="rocketIMG" />
    <div className="rocket-info">
      <h2>{name}</h2>
      <p>{description}</p>
      <button type="button">Reserve rocket</button>
    </div>
  </div>
);

Rocket.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};
export default Rocket;
