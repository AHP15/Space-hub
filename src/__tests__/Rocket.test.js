import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider, useSelector } from 'react-redux';
import PropTypes from 'prop-types';

import Rocket from '../components/Rocket';
import Store from '../slices/store';

it('renders Rocket correctly', () => {
  const tree = renderer.create(
    <Provider store={Store}>
      <Rocket
        name="Falcon 1"
        image="rocket image"
        description="Rocket description"
        id={1}
        reserved={false}
      />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

const CustomizedCompenent = ({ index }) => {
  const { rockets } = useSelector((state) => state.rockets);
  return (
    <section>
      <div>
        <Rocket
          key={rockets[index].id}
          name={rockets[index].rocket_name}
          image={rockets[index].flickr_images[0]}
          description={rockets[index].description}
          id={rockets[index].id}
          reserved={rockets[index].reserved}
        />
      </div>
    </section>
  );
};

CustomizedCompenent.propTypes = {
  index: PropTypes.number.isRequired,
};

it('Join button works correctly', async () => {
  // Arrange
  render(<Provider store={Store}><CustomizedCompenent index={0} /></Provider>);

  // Act
  await userEvent.click(screen.getByText('Cancel reservation'));
  await screen.getByTitle('isMember');

  // Assertion
  expect(screen.getByTitle('isMember')).toHaveTextContent('Reserve rocket');
});

it('Leave button works correctly', async () => {
  // Arrange
  render(<Provider store={Store}><CustomizedCompenent index={1} /></Provider>);

  // Act
  await userEvent.click(screen.getByText('Reserve rocket'));
  await screen.getByTitle('isMember');

  // Assertion
  expect(screen.getByTitle('isMember')).toHaveTextContent('Cancel reservation');
});
