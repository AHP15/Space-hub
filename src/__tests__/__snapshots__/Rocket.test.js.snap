// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Rocket correctly 1`] = `
<div
  className="card"
>
  <img
    alt="rocketIMG"
    src="rocket image"
  />
  <div
    className="rocket-info"
  >
    <h2>
      Falcon 1
    </h2>
    <p>
      Rocket description
    </p>
    <button
      className=""
      onClick={[Function]}
      title="isMember"
      type="button"
    >
      Reserve rocket
    </button>
  </div>
</div>
`;
